tests/ui/coherence/occurs-check/associated-type2.rs

```diff
@@ -0,0 +1,20 @@
+// revisions: old next
+//[next] compile-flags: -Znext-solver
```

Comment:
please add a comment and further minimize this test. rn it's confusing what exactly this is testing. This test is unrelated to the occurs check, so pls also move it to a different folder

```suggestion
// A regression test for #120343. The overlap error was previously
// silenced in coherence because projecting `<() as ToUnit>::Unit`
// failed. Then then silenced the missing items error in the `ToUnit`
// impl, causing us to not emit any errors and ICEing due to a
// `span_delay_bug`.
trait ToUnit {
    type Unit;
}

impl ToUnit for () {}

trait Overlap {}
impl<T> Overlap for T {}
impl Overlap for <() as ToUnit>::Unit {}

fn main() {}
```